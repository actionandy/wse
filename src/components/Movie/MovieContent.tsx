import Movie, {MovieData} from './Movie';
import React from 'react';
import MovieList from './MovieList';

interface MovieContentProps {
    movies: MovieData[];
    movieResponse: MovieData[];
    loadMoreMovies: boolean;
    hasMoreMovies: (node: HTMLElement | null) => any;
    offset: number;
    limit: number;
}

const MovieContent = ({movies, loadMoreMovies, movieResponse, hasMoreMovies, offset, limit}: MovieContentProps) => (movies.length ? <>
  {movies.map((movie) => <Movie key={movie.ProviderTrackID} {...movie} />)}
  {!loadMoreMovies && <div ref={hasMoreMovies}></div>}
  {movieResponse.length > limit && loadMoreMovies && <MovieList offset={offset + limit} limit={limit} />}
</> : <>Vorschl√§ge werden geladen...</>);

export default MovieContent;
