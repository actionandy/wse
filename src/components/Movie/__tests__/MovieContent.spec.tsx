/* eslint-disable require-jsdoc */
import MovieContent from '../MovieContent';
import {render} from '@testing-library/react';
import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import movies from './__mocks__/movies.json';

(global as any).IntersectionObserver = class IntersectionObserver {
  constructor() {}

  observe() {
    return null;
  }

  disconnect() {
    return null;
  };

  unobserve() {
    return null;
  }
};

describe('MovieContent', () => {
  it('RenderLoadingTestWhenNoMoviesProvided', () => {
    const {container} = render(<MovieContent movies={[]} loadMoreMovies={false} movieResponse={[]} hasMoreMovies={() => {}} offset={0} limit={36} />);
    expect(container).toMatchSnapshot();
  });

  it('RenderLoadingTestWhenMoviesAreProvided', () => {
    const {container} = render(<BrowserRouter>
      <MovieContent movies={movies} loadMoreMovies={false} movieResponse={[]} hasMoreMovies={() => {}} offset={0} limit={36} />
    </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
