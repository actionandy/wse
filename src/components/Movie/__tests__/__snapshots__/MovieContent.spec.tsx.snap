// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContent RenderLoadingTestWhenMoviesAreProvided 1`] = `
<div>
  <div
    class="relative"
  >
    <div
      class="relative hover:cursor-pointer"
    >
      <a
        href="/filme/-9YZA_G7GJ0.P/gunda"
        title="Weitere Informationen und Trailer zu Gunda"
      >
        <img
          alt="Filmposter: Gunda"
          class="w-fulll rounded border-2 p-1 shadow-sm"
          referrerpolicy="no-referrer"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg=="
          style="height: 326px; width: 230px;"
        />
        <div
          class="overflow-hidden absolute top-0 right-0 bottom-0 left-0 w-full h-full bg-fixed opacity-0 hover:opacity-40 hover:bg-yellow-300 transition duration-300 ease-in-out bg-white"
        />
      </a>
    </div>
  </div>
  <div
    class="relative"
  >
    <div
      class="relative hover:cursor-pointer"
    >
      <a
        href="/filme/AOYloSgV5MA.P/kevin-allein-zu-haus"
        title="Weitere Informationen und Trailer zu Kevin - Allein zu Haus"
      >
        <img
          alt="Filmposter: Kevin - Allein zu Haus"
          class="w-fulll rounded border-2 p-1 shadow-sm"
          referrerpolicy="no-referrer"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg=="
          style="height: 326px; width: 230px;"
        />
        <div
          class="overflow-hidden absolute top-0 right-0 bottom-0 left-0 w-full h-full bg-fixed opacity-0 hover:opacity-40 hover:bg-yellow-300 transition duration-300 ease-in-out bg-white"
        />
      </a>
    </div>
  </div>
  <div
    class="relative"
  >
    <div
      class="relative hover:cursor-pointer"
    >
      <a
        href="/filme/t1uJ0wSV2_U.P/dune-drifter"
        title="Weitere Informationen und Trailer zu Dune Drifter"
      >
        <img
          alt="Filmposter: Dune Drifter"
          class="w-fulll rounded border-2 p-1 shadow-sm"
          referrerpolicy="no-referrer"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg=="
          style="height: 326px; width: 230px;"
        />
        <div
          class="overflow-hidden absolute top-0 right-0 bottom-0 left-0 w-full h-full bg-fixed opacity-0 hover:opacity-40 hover:bg-yellow-300 transition duration-300 ease-in-out bg-white"
        />
      </a>
    </div>
  </div>
  <div />
</div>
`;

exports[`MovieContent RenderLoadingTestWhenNoMoviesProvided 1`] = `
<div>
  Vorschl√§ge werden geladen...
</div>
`;
